// mindee/src/product/us/driverLicense/driverLicenseV1Page.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DriverLicenseV1Page = void 0;
const common_1 = require("../../../parsing/common");
const standard_1 = require("../../../parsing/standard");
const driverLicenseV1Document_1 = require("./driverLicenseV1Document");
/**
 * Page data for Driver License, API version 1.
 */
class DriverLicenseV1Page extends driverLicenseV1Document_1.DriverLicenseV1Document {
    constructor(rawPrediction, pageId) {
        super(rawPrediction, pageId);
        this.photo = new standard_1.PositionField({
            prediction: rawPrediction["photo"],
            pageId: pageId,
        });
        this.signature = new standard_1.PositionField({
            prediction: rawPrediction["signature"],
            pageId: pageId,
        });
    }
    toString() {
        let outStr = `:Photo: ${this.photo}
:Signature: ${this.signature}`.trimEnd();
        outStr += "\n" + super.toString();
        return (0, common_1.cleanOutString)(outStr);
    }
}
exports.DriverLicenseV1Page = DriverLicenseV1Page;
