// mindee/src/product/us/driverLicense/driverLicenseV1Document.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DriverLicenseV1Document = void 0;
const common_1 = require("../../../parsing/common");
const standard_1 = require("../../../parsing/standard");
/**
 * Document data for Driver License, API version 1.
 */
class DriverLicenseV1Document {
    constructor(rawPrediction, pageId) {
        this.address = new standard_1.StringField({
            prediction: rawPrediction["address"],
            pageId: pageId,
        });
        this.dateOfBirth = new standard_1.DateField({
            prediction: rawPrediction["date_of_birth"],
            pageId: pageId,
        });
        this.ddNumber = new standard_1.StringField({
            prediction: rawPrediction["dd_number"],
            pageId: pageId,
        });
        this.dlClass = new standard_1.StringField({
            prediction: rawPrediction["dl_class"],
            pageId: pageId,
        });
        this.driverLicenseId = new standard_1.StringField({
            prediction: rawPrediction["driver_license_id"],
            pageId: pageId,
        });
        this.endorsements = new standard_1.StringField({
            prediction: rawPrediction["endorsements"],
            pageId: pageId,
        });
        this.expiryDate = new standard_1.DateField({
            prediction: rawPrediction["expiry_date"],
            pageId: pageId,
        });
        this.eyeColor = new standard_1.StringField({
            prediction: rawPrediction["eye_color"],
            pageId: pageId,
        });
        this.firstName = new standard_1.StringField({
            prediction: rawPrediction["first_name"],
            pageId: pageId,
        });
        this.hairColor = new standard_1.StringField({
            prediction: rawPrediction["hair_color"],
            pageId: pageId,
        });
        this.height = new standard_1.StringField({
            prediction: rawPrediction["height"],
            pageId: pageId,
        });
        this.issuedDate = new standard_1.DateField({
            prediction: rawPrediction["issued_date"],
            pageId: pageId,
        });
        this.lastName = new standard_1.StringField({
            prediction: rawPrediction["last_name"],
            pageId: pageId,
        });
        this.restrictions = new standard_1.StringField({
            prediction: rawPrediction["restrictions"],
            pageId: pageId,
        });
        this.sex = new standard_1.StringField({
            prediction: rawPrediction["sex"],
            pageId: pageId,
        });
        this.state = new standard_1.StringField({
            prediction: rawPrediction["state"],
            pageId: pageId,
        });
        this.weight = new standard_1.StringField({
            prediction: rawPrediction["weight"],
            pageId: pageId,
        });
    }
    /**
     * Default string representation.
     */
    toString() {
        const outStr = `:State: ${this.state}
:Driver License ID: ${this.driverLicenseId}
:Expiry Date: ${this.expiryDate}
:Date Of Issue: ${this.issuedDate}
:Last Name: ${this.lastName}
:First Name: ${this.firstName}
:Address: ${this.address}
:Date Of Birth: ${this.dateOfBirth}
:Restrictions: ${this.restrictions}
:Endorsements: ${this.endorsements}
:Driver License Class: ${this.dlClass}
:Sex: ${this.sex}
:Height: ${this.height}
:Weight: ${this.weight}
:Hair Color: ${this.hairColor}
:Eye Color: ${this.eyeColor}
:Document Discriminator: ${this.ddNumber}`.trimEnd();
        return (0, common_1.cleanOutString)(outStr);
    }
}
exports.DriverLicenseV1Document = DriverLicenseV1Document;
