// mindee/src/cli.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.cli = void 0;
const commander_1 = require("commander");
const product = __importStar(require("./product"));
const client_1 = require("./client");
const input_1 = require("./input");
const console = __importStar(require("console"));
const program = new commander_1.Command();
const COMMAND_CUSTOM = "custom";
//
// PRODUCT CONFIGURATION
//
// The Map's key is the command name as it will appear on the console.
//
const CLI_COMMAND_CONFIG = new Map([
    [
        COMMAND_CUSTOM,
        {
            displayName: "Custom Document",
            docClass: product.CustomV1,
            allWords: false,
            async: false,
            sync: true,
        },
    ],
    [
        "cropper",
        {
            displayName: "Cropper",
            docClass: product.CropperV1,
            allWords: false,
            async: false,
            sync: true,
        },
    ],
    [
        "invoice",
        {
            displayName: "Invoice",
            docClass: product.InvoiceV4,
            allWords: true,
            async: false,
            sync: true,
        },
    ],
    [
        "invoice-splitter",
        {
            displayName: "Invoice Splitter",
            docClass: product.InvoiceSplitterV1,
            allWords: false,
            async: true,
            sync: false,
        },
    ],
    [
        "receipt",
        {
            displayName: "Expense Receipt",
            docClass: product.ReceiptV5,
            allWords: true,
            async: false,
            sync: true,
        },
    ],
    [
        "passport",
        {
            displayName: "Passport",
            docClass: product.PassportV1,
            allWords: false,
            async: false,
            sync: true,
        },
    ],
    [
        "financial",
        {
            displayName: "Financial Document",
            docClass: product.FinancialDocumentV1,
            allWords: true,
            async: false,
            sync: true,
        },
    ],
    [
        "fr-id-card",
        {
            displayName: "FR ID Card",
            docClass: product.fr.IdCardV1,
            allWords: false,
            async: false,
            sync: true,
        },
    ],
    [
        "fr-bank-account-details",
        {
            displayName: "FR Bank Account Details",
            docClass: product.fr.BankAccountDetailsV2,
            allWords: false,
            async: false,
            sync: true,
        },
    ],
    [
        "fr-carte-vitale",
        {
            displayName: "FR Carte Vitale",
            docClass: product.fr.CarteVitaleV1,
            allWords: false,
            async: false,
            sync: true,
        },
    ],
    [
        "eu-license-plate",
        {
            displayName: "EU License Plate",
            docClass: product.eu.LicensePlateV1,
            allWords: false,
            async: false,
            sync: true,
        },
    ],
    [
        "us-bank-check",
        {
            displayName: "US Bank Check",
            docClass: product.us.BankCheckV1,
            allWords: false,
            async: false,
            sync: true,
        },
    ],
]);
//
// EXECUTE THE COMMANDS
//
function initClient(options) {
    const mindeeClient = new client_1.Client({
        apiKey: options.apiKey,
        debug: options.debug,
    });
    return mindeeClient;
}
function getConfig(command) {
    const conf = CLI_COMMAND_CONFIG.get(command);
    if (conf === undefined) {
        throw new Error(`Invalid document type ${command}`);
    }
    return conf;
}
function getPageOptions(options) {
    let pageOptions = undefined;
    if (options.cutPages) {
        pageOptions = {
            operation: input_1.PageOptionsOperation.KeepOnly,
            pageIndexes: [0, 1, 2, 3, 4],
            onMinPages: 5,
        };
    }
    return pageOptions;
}
function getPredictParams(options) {
    const predictParams = {
        allWords: options.allWords,
        cropper: options.cropper,
    };
    return predictParams;
}
async function callParse(command, inputPath, options) {
    const conf = getConfig(command);
    const mindeeClient = initClient(options);
    const predictParams = getPredictParams(options);
    const pageOptions = getPageOptions(options);
    const inputSource = mindeeClient.docFromPath(inputPath);
    let response;
    if (command === COMMAND_CUSTOM) {
        const customEndpoint = mindeeClient.createEndpoint(options.endpoint, options.account, options.version);
        response = await mindeeClient.parse(conf.docClass, inputSource, {
            endpoint: customEndpoint,
            pageOptions: pageOptions,
            allWords: predictParams.allWords,
            cropper: predictParams.cropper,
        });
    }
    else {
        response = await mindeeClient.parse(conf.docClass, inputSource, {
            pageOptions: pageOptions,
            allWords: predictParams.allWords,
            cropper: predictParams.cropper,
        });
    }
    printResponse(response.document, options);
}
async function callEnqueue(command, inputPath, options) {
    const conf = getConfig(command);
    const mindeeClient = initClient(options);
    const predictParams = getPredictParams(options);
    const pageOptions = getPageOptions(options);
    const inputSource = mindeeClient.docFromPath(inputPath);
    let response;
    if (command === COMMAND_CUSTOM) {
        const customEndpoint = mindeeClient.createEndpoint(options.endpoint, options.account, options.version);
        response = await mindeeClient.enqueue(conf.docClass, inputSource, {
            endpoint: customEndpoint,
            pageOptions: pageOptions,
            allWords: predictParams.allWords,
            cropper: predictParams.cropper,
        });
    }
    else {
        response = await mindeeClient.enqueue(conf.docClass, inputSource, {
            pageOptions: pageOptions,
            allWords: predictParams.allWords,
            cropper: predictParams.cropper,
        });
    }
    console.log(response.job);
}
async function callParseQueued(command, queueId, options) {
    const conf = getConfig(command);
    const mindeeClient = initClient(options);
    const response = await mindeeClient.parseQueued(conf.docClass, queueId);
    if (response.document !== undefined) {
        printResponse(response.document, options);
    }
    else {
        console.log(response.job);
    }
}
function printResponse(document, options) {
    if (options.allWords) {
        document.ocr?.mVisionV1.pages.forEach((page) => {
            console.log(page.allWords.toString());
        });
    }
    if (options.pages) {
        document.inference.pages.forEach((page) => {
            console.log(`\n${page}`);
        });
    }
    if (document) {
        console.log(`\n${document}`);
    }
}
//
// BUILD THE COMMANDS
//
function addMainOptions(prog) {
    prog.option("-k, --api-key <api_key>", "API key for document endpoint");
}
function addPostOptions(prog, info) {
    prog.option("-c, --cut-pages", "keep only the first 5 pages of the document");
    if (info.allWords) {
        prog.option("-w, --all-words", "to get all the words in the current document. False by default.");
    }
    prog.argument("<input_path>", "full path to the file");
}
function addCustomPostOptions(prog) {
    prog.requiredOption("-e, --endpoint <endpoint_name>", "API endpoint name (required)");
    prog.requiredOption("-a, --account <account_name>", "API account name for the endpoint (required)");
    prog.option("-v, --version <version>", "version for the endpoint, use the latest version if not specified");
}
function addDisplayOptions(prog) {
    prog.option("-p, --pages", "show content of individual pages");
}
function routeSwitchboard(command, inputPath, allOptions) {
    switch (command.parent?.name()) {
        case "parse": {
            return callParse(command.name(), inputPath, allOptions);
        }
        case "enqueue": {
            return callEnqueue(command.name(), inputPath, allOptions);
        }
        case "parse-queued": {
            return callParseQueued(command.name(), inputPath, allOptions);
        }
        default: {
            throw new Error("Unhandled parent command.");
        }
    }
}
function addAction(prog) {
    if (prog.name() === COMMAND_CUSTOM) {
        prog.action(function (inputPath, options, command) {
            const allOptions = {
                ...prog.parent?.parent?.opts(),
                ...prog.parent?.opts(),
                ...prog.opts(),
                ...options,
            };
            return routeSwitchboard(command, inputPath, allOptions);
        });
    }
    else {
        prog.action(function (inputPath, options, command) {
            const allOptions = {
                ...prog.parent?.parent?.opts(),
                ...prog.parent?.opts(),
                ...prog.opts(),
                ...options,
            };
            return routeSwitchboard(command, inputPath, allOptions);
        });
    }
}
function cli() {
    program.name("mindee");
    program.option("-d, --debug", "high verbosity mode");
    const predict = program.command("parse").description("Parse synchronously.");
    addMainOptions(predict);
    const enqueue = program
        .command("enqueue")
        .description("Add to async parse queue.");
    addMainOptions(enqueue);
    const parseQueued = program
        .command("parse-queued")
        .description("Parse from async queue.");
    addMainOptions(parseQueued);
    CLI_COMMAND_CONFIG.forEach((info, name) => {
        if (info.sync) {
            const prog = predict
                .command(name)
                .description(`Parse an ${info.displayName}.`);
            if (name === COMMAND_CUSTOM) {
                addCustomPostOptions(prog);
            }
            addDisplayOptions(prog);
            addPostOptions(prog, info);
            addAction(prog);
        }
        if (info.async) {
            const progEnqueue = enqueue
                .command(name)
                .description(`Add an ${info.displayName} to the queue.`);
            if (name === COMMAND_CUSTOM) {
                addCustomPostOptions(progEnqueue);
            }
            addPostOptions(progEnqueue, info);
            addAction(progEnqueue);
            const progParse = parseQueued
                .command(name)
                .description(`Parse an ${info.displayName} from the queue.`)
                .argument("<doc_id>", "ID of the document");
            addDisplayOptions(progParse);
            addAction(progParse);
        }
    });
    program.parse(process.argv);
}
exports.cli = cli;
