// mindee/src/parsing/standard/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Field = exports.StringField = exports.PositionField = exports.CompanyRegistrationField = exports.BaseField = exports.DateField = exports.floatToString = exports.AmountField = exports.LocaleField = exports.PaymentDetailsField = exports.TaxField = exports.Taxes = exports.ClassificationField = void 0;
var classification_1 = require("./classification");
Object.defineProperty(exports, "ClassificationField", { enumerable: true, get: function () { return classification_1.ClassificationField; } });
var tax_1 = require("./tax");
Object.defineProperty(exports, "Taxes", { enumerable: true, get: function () { return tax_1.Taxes; } });
Object.defineProperty(exports, "TaxField", { enumerable: true, get: function () { return tax_1.TaxField; } });
var paymentDetails_1 = require("./paymentDetails");
Object.defineProperty(exports, "PaymentDetailsField", { enumerable: true, get: function () { return paymentDetails_1.PaymentDetailsField; } });
var locale_1 = require("./locale");
Object.defineProperty(exports, "LocaleField", { enumerable: true, get: function () { return locale_1.LocaleField; } });
var amount_1 = require("./amount");
Object.defineProperty(exports, "AmountField", { enumerable: true, get: function () { return amount_1.AmountField; } });
Object.defineProperty(exports, "floatToString", { enumerable: true, get: function () { return amount_1.floatToString; } });
var date_1 = require("./date");
Object.defineProperty(exports, "DateField", { enumerable: true, get: function () { return date_1.DateField; } });
var base_1 = require("./base");
Object.defineProperty(exports, "BaseField", { enumerable: true, get: function () { return base_1.BaseField; } });
var companyRegistration_1 = require("./companyRegistration");
Object.defineProperty(exports, "CompanyRegistrationField", { enumerable: true, get: function () { return companyRegistration_1.CompanyRegistrationField; } });
var position_1 = require("./position");
Object.defineProperty(exports, "PositionField", { enumerable: true, get: function () { return position_1.PositionField; } });
var text_1 = require("./text");
Object.defineProperty(exports, "StringField", { enumerable: true, get: function () { return text_1.StringField; } });
var field_1 = require("./field");
Object.defineProperty(exports, "Field", { enumerable: true, get: function () { return field_1.Field; } });
